Engine/OGF-Core/Core/Graphics/Texture.h

```diff
@@ -40,16 +64,22 @@ class Texture
 	};
 
 private:
-	GLuint m_name;
-	Target m_target;
-	Texture(Target target);
+	GLuint _texid;
+	Target _target;
+	InternalFormat _internalFormat;
+	Format _format;
+	Type _type;
+
+	Texture(Target target, InternalFormat internalFormat, Format format, Type type);
 public:
 	virtual ~Texture();
 
 	void bind(Unit unit) const;
 	void unbind() const;
+	void resize(const int32 width, const int32 height);
 
-	Target getTarget() const { return m_target; };
+	Target getTarget() const { return _target; };
+	GLuint getId() const { return _texid; };
```

Comment:
minor: try to use engine typedefs over opengl ones as much as possible, it will help once we decide to implement different renderers such as vulkan or dx12